tests/ops/functions/test_generator.py

```diff
@@ -341,22 +341,38 @@ class TestObservableReturn:
     """Tests for format="observable". This format preserves the initial generator
     encoded in the operator."""
 
+    @pytest.mark.usefixtures("use_legacy_opmath")
     def test_observable(self):
         """Test a generator that returns a single observable is correct"""
         gen = qml.generator(ObservableOp, format="observable")(0.5, wires=0)
         assert gen.name == "Hamiltonian"
         assert gen.compare(ObservableOp(0.5, wires=0).generator())
 
+    def test_observable_opmath(self):
+        """Test a generator that returns a single observable is correct with opmath enabled"""
+        gen = qml.generator(ObservableOp, format="observable")(0.5, wires=0)
+        assert gen.name == "SProd"
+        assert qml.equal(gen, ObservableOp(0.5, wires=0).generator())
+
+    @pytest.mark.usefixtures("use_legacy_opmath")
     def test_tensor_observable(self):
         """Test a generator that returns a tensor observable is correct"""
         gen = qml.generator(TensorOp, format="observable")(0.5, wires=[0, 1])
         assert gen.name == "Hamiltonian"
         assert gen.compare(TensorOp(0.5, wires=[0, 1]).generator())
 
+    def test_tensor_observable_opmath(self):
+        """Test a generator that returns a tensor observable is correct with opmath enabled"""
+        gen = qml.generator(TensorOp, format="observable")(0.5, wires=[0, 1])
+        assert gen.name == "Prod"
+        assert qml.equal(gen, TensorOp(0.5, wires=[0, 1]).generator())
+
+    # Also works with opmath disabled by replacing
+    # assert gen.name == "LinearCombination"   with   assert gen.name == "Hamiltonian"
     def test_hamiltonian(self):
         """Test a generator that returns a Hamiltonian"""
         gen = qml.generator(HamiltonianOp, format="observable")(0.5, wires=[0, 1])
-        assert gen.name == "Hamiltonian"
+        assert gen.name == "LinearCombination"

```

Comment:
I believe this could pass both ways if it was 

`assert gen.name == qml.Hamiltonian([], []).name` 

or 

```
expected_type = type(qml.Hamiltonian([],[]))
assert isinstance(gen, expected_type)
```

But not sure how much that matters.

Reply:
True! I'm implementing that. Thanks for the suggestion